### PR TITLE
get_app_icon_html and tests

### PR DESCRIPTION
**Pull request by gromdimon**
Task #1057 get_app_icon_html().
 - Changed a little bit timeline/templatetags/timeline_tags.py
 - Added 2 tests in timeline/tests/test_templatetsgs.py
 ---

Will be happy for any feedback

> Was: Screenshot from 2022-12-08 15-47-22
> Became: Screenshot from 2022-12-08 16-47-41

P.s.
I've also made changes on another tasks at my branch, so I hope, I do not push other files.

### CODE REVIEW

File: timeline/tests/test_templatetags.py [hunk 1]
```diff
@@ -180,6 +180,39 @@ def test_get_app_icon_html_plugin(self):
...
+        url = reverse(
+            plugin.entry_point_url_id,
+            kwargs={'project': self.project.sodar_uuid},
+        )
+        ret = tags.get_app_icon_html(event, self.plugin_lookup)
+        self.assertIn(plugin.icon, ret)
+        self.assertIn(plugin.title, ret)
+        self.assertIn(url, ret)
+
+    def test_get_app_icon_html_timeline(self):
```
Comment (mikkonie) [inline timeline/tests/test_templatetags.py]: We can omit this testcase, as the timeline app itself is not expected to create any timeline events.

File: timeline/tests/test_templatetags.py [hunk 2]
```diff
@@ -180,6 +180,39 @@ def test_get_app_icon_html_plugin(self):
...
+        )
+        plugin = get_app_plugin('example_site_app')
+        url = reverse(
+            plugin.entry_point_url_id,
+            kwargs={'project': self.project.sodar_uuid},
+        )
+        ret = tags.get_app_icon_html(event, self.plugin_lookup)
+        self.assertIn(plugin.icon, ret)
+        self.assertIn(plugin.title, ret)
+        self.assertIn(url, ret)
```
Comment (mikkonie) [inline timeline/tests/test_templatetags.py:200]: We should also assert that `url` corresponds to the url reversed from the `entry_point_url_id` of the plugin. Other than that the test case looks very good.

File: timeline/templatetags/timeline_tags.py [hunk 3]
```diff
@@ -86,8 +86,15 @@ def get_app_icon_html(event, plugin_lookup):
         if plugin_name in plugin_lookup.keys():
             plugin = plugin_lookup[plugin_name]
             entry_point = getattr(plugin, 'entry_point_url_id', None)
-            if entry_point:
-                url = reverse(entry_point, kwargs=url_kwargs)
+            if entry_point in plugin_lookup.items:
```
Comment (mikkonie) [inline timeline/templatetags/timeline_tags.py:94]: This try-except is good as it fixes the immediate problem of the crash.

We still need to have the check for the plugin type, as specified in the issue. ONLY provide `project` in `url_kwargs`, if the if the plugin is based on `ProjectAppPlugin`. Site and backend apps will not have the project kwarg in their entry point URL.

**Comment by gromdimon** [comment]
I've just changed two files and wanted to update this pull request, but I closed it somehow )(
I hope, it's allright

**Pull request merged**